scripts/src/main/resources/scripts/command/docker

```diff
@@ -16,46 +16,125 @@ source "$(dirname "${0}")"/../functions
 
 # $1: optional setup
 function doSetup() {  
-  if doIsDockerInstalled 
+  if [ "${DOCKER_EDITION}" != "DOCKERDESKTOP" ]
   then
-    if [ "${1}" != "silent" ] && ! doIsQuiet
-    then
-      doEcho "docker is already installed at $(command -v docker)"
-      docker version
+    echo "Rancher Desktop set as option"
+    if uname -r | grep -i microsoft
+    then 
+      echo "It seems that you are running the setup from inside a WSL shell."
+      echo "Please use git-bash for that purpose. Otherwise your OS will be interpreted as Linux"
+    else
+      if doIsRancherInstalled 
+      then
+        doIsWindows 
+        queryWin=$?
+        if [ "${1}" != "silent" ] && ! doIsQuiet
+        then
+          #give the User the installation location when using linux or mac
+          if [ $queryWin != 0 ]
+          then
+            doEcho "Rancher Desktop is already installed at $(command -v devon docker)"
+          else
+            doEcho "Rancher Desktop is already installed"
+          fi
+        fi
+      else
+        doRequireNotBatch
+        local latestversion
+        latestversion=$(curl -Ls -o /dev/null -w '%{url_effective}' https://github.com/rancher-sandbox/rancher-desktop/releases/latest | sed 's/.*\([0-9]\.[0-9]\.[0-9]\).*/\1/')
+        local previousversion="1.2.1"
+        if [[ "${latestversion}" =~ ^[0-9]\.[0-9].[0-9]$ ]] 
+        then
+          if [ "${latestversion}" \> "${previousversion}" ]
+          then
+            doEcho "setting RANCHER_VERSION"
+            RANCHER_VERSION=${latestversion}
+            doEcho "RANCHER_VERSION is ${RANCHER_VERSION}"
+          fi
+        fi
+        local version="${RANCHER_VERSION:-1.2.1}"
```

Comment:
Version `1.2.1` gets duplicated here with `previousversion` in line 45.
You could avoid this redundancy.
Further the code above is inefficient as the latest version is always determined causing `curl` and `http` remote invocations but if `${RANCER_VERSION}` is set the latestversion and the result of the version-comparison is always discarded.
So IMHO simply do this:
```
local version="${RANCHER_VERSION}"
if [ -z "${version}" ]
then
  version=1.2.1
  # ... determine latest version and update version if newer ...
fi
```

BTW: I still do not get what we need `1.2.1` for if we simply get the latest version from CURL. So if CURL fails we use `1.2.1` as fallback, right?

Reply:
Also solved. Thanks.